NX 19.4.2 - Build Command Fails After Creating Repository with Angular and NX

### Current Behavior

I ran the command to create a repository with Angular and NX. As soon as it finished, I tried running the build command, and it failed:

```
npx create-nx-workspace@latest --name nx-ng-build-problem --pm yarn

 NX   Let's create a new workspace [https://nx.dev/getting-started/intro]

✔ Which stack do you want to use? · angular
✔ Integrated monorepo, or standalone project? · integrated
✔ Application name · nx-ng-build-problem
✔ Which bundler would you like to use? · esbuild
✔ Default stylesheet format · scss
✔ Do you want to enable Server-Side Rendering (SSR) and Static Site Generation (SSG/Prerendering)? · No
✔ Test runner to use for end to end (E2E) tests · playwright
✔ Which CI provider would you like to use? · skip
✔ Would you like remote caching to make your build faster? · skip

 NX   Creating your v19.4.2 workspace.
```

Then:
```
yarn nx run-many -t build,lint,test --skip-nx-cache
yarn run v1.22.22
$ /Users/wizardnet972/code/nx-ng-build-problem/node_modules/.bin/nx run-many -t build,lint,test --skip-nx-cache

 NX   Unable to resolve @angular-devkit/build-angular:application.

Cannot find module '@angular/build/package.json'
Require stack:
- /Users/wizardnet972/code/nx-ng-build-problem/node_modules/nx/src/utils/package-json.js
- /Users/wizardnet972/code/nx-ng-build-problem/node_modules/nx/src/project-graph/plugins/loader.js
- /Users/wizardnet972/code/nx-ng-build-problem/node_modules/nx/src/project-graph/plugins/index.js
- /Users/wizardnet972/code/nx-ng-build-problem/node_modules/nx/src/command-line/run/executor-utils.js
- /Users/wizardnet972/code/nx-ng-build-problem/node_modules/nx/src/devkit-internals.js
- /Users/wizardnet972/code/nx-ng-build-problem/node_modules/nx/src/utils/assert-workspace-validity.js
- /Users/wizardnet972/code/nx-ng-build-problem/node_modules/nx/src/project-graph/build-project-graph.js
- /Users/wizardnet972/code/nx-ng-build-problem/node_modules/nx/src/project-graph/project-graph.js
- /Users/wizardnet972/code/nx-ng-build-problem/node_modules/nx/src/project-graph/file-utils.js
- /Users/wizardnet972/code/nx-ng-build-problem/node_modules/nx/src/utils/package-manager.js
- /Users/wizardnet972/code/nx-ng-build-problem/node_modules/nx/src/utils/child-process.js
- /Users/wizardnet972/code/nx-ng-build-problem/node_modules/nx/src/command-line/migrate/command-object.js
- /Users/wizardnet972/code/nx-ng-build-problem/node_modules/nx/src/command-line/nx-commands.js
- /Users/wizardnet972/code/nx-ng-build-problem/node_modules/nx/bin/init-local.js
- /Users/wizardnet972/code/nx-ng-build-problem/node_modules/nx/bin/nx.js
Pass --verbose to see the stacktrace.

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### Expected Behavior

The build should have succeeded.

### GitHub Repo

https://github.com/wizardnet972/nx-ng-build-problem

### Steps to Reproduce

see github repo or follow the code above.

### Nx Report

```shell
yarn nx report
yarn run v1.22.22
$ /Users/wizardnet972/code/nx-ng-build-problem/node_modules/.bin/nx report

 NX   Report complete - copy this into the issue template

Node   : 20.14.0
OS     : darwin-arm64
yarn   : 1.22.22

nx                 : 19.4.2
@nx/js             : 19.4.2
@nx/jest           : 19.4.2
@nx/linter         : 19.4.2
@nx/eslint         : 19.4.2
@nx/workspace      : 19.4.2
@nx/angular        : 19.4.2
@nx/devkit         : 19.4.2
@nx/eslint-plugin  : 19.4.2
@nx/playwright     : 19.4.2
@nrwl/tao          : 19.4.2
@nx/web            : 19.4.2
@nx/webpack        : 19.4.2
typescript         : 5.4.5
---------------------------------------
Registered Plugins:
@nx/playwright/plugin
@nx/eslint/plugin

✨  Done in 0.66s.
```
```


### Failure Logs

_No response_

### Package Manager Version

_No response_

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

_No response_

--- a/packages/nx/src/adapter/ngcli-adapter.ts
+++ b/packages/nx/src/adapter/ngcli-adapter.ts
@@ -1190,7 +1190,8 @@ async function getWrappedWorkspaceNodeModulesArchitectHost(
 
     private readExecutorsJson(
       nodeModule: string,
-      builder: string
+      builder: string,
+      extraRequirePaths: string[] = []
     ): {
       executorsFilePath: string;
       executorConfig: ExecutorJsonEntryConfig;
@@ -1200,7 +1201,9 @@ async function getWrappedWorkspaceNodeModulesArchitectHost(
         readPluginPackageJson(
           nodeModule,
           this.projects,
-          this.root ? [this.root, __dirname] : [__dirname]
+          this.root
+            ? [this.root, __dirname, ...extraRequirePaths]
+            : [__dirname, ...extraRequirePaths]
         );
       const executorsFile = packageJson.executors ?? packageJson.builders;
 
@@ -1210,9 +1213,8 @@ async function getWrappedWorkspaceNodeModulesArchitectHost(
         );
       }
 
-      const executorsFilePath = require.resolve(
-        join(dirname(packageJsonPath), executorsFile)
-      );
+      const basePath = dirname(packageJsonPath);
+      const executorsFilePath = require.resolve(join(basePath, executorsFile));
       const executorsJson = readJsonFile<ExecutorsJson>(executorsFilePath);
       const executorConfig =
         executorsJson.builders?.[builder] ?? executorsJson.executors?.[builder];
@@ -1224,7 +1226,7 @@ async function getWrappedWorkspaceNodeModulesArchitectHost(
       if (typeof executorConfig === 'string') {
         // Angular CLI can have a builder pointing to another package:builder
         const [packageName, executorName] = executorConfig.split(':');
-        return this.readExecutorsJson(packageName, executorName);
+        return this.readExecutorsJson(packageName, executorName, [basePath]);
       }
 
       return { executorsFilePath, executorConfig, isNgCompat: true };

--- a/packages/nx/src/command-line/run/executor-utils.ts
+++ b/packages/nx/src/command-line/run/executor-utils.ts
@@ -99,7 +99,8 @@ function readExecutorJson(
   nodeModule: string,
   executor: string,
   root: string,
-  projects: Record<string, ProjectConfiguration>
+  projects: Record<string, ProjectConfiguration>,
+  extraRequirePaths: string[] = []
 ): {
   executorsFilePath: string;
   executorConfig: {
@@ -114,8 +115,14 @@ function readExecutorJson(
     nodeModule,
     projects,
     root
-      ? [root, __dirname, process.cwd(), ...getNxRequirePaths()]
-      : [__dirname, process.cwd(), ...getNxRequirePaths()]
+      ? [
+          root,
+          __dirname,
+          process.cwd(),
+          ...getNxRequirePaths(),
+          ...extraRequirePaths,
+        ]
+      : [__dirname, process.cwd(), ...getNxRequirePaths(), ...extraRequirePaths]
   );
   const executorsFile = packageJson.executors ?? packageJson.builders;
 
@@ -125,9 +132,8 @@ function readExecutorJson(
     );
   }
 
-  const executorsFilePath = require.resolve(
-    join(dirname(packageJsonPath), executorsFile)
-  );
+  const basePath = dirname(packageJsonPath);
+  const executorsFilePath = require.resolve(join(basePath, executorsFile));
   const executorsJson = readJsonFile<ExecutorsJson>(executorsFilePath);
   const executorConfig =
     executorsJson.executors?.[executor] || executorsJson.builders?.[executor];
@@ -139,7 +145,9 @@ function readExecutorJson(
   if (typeof executorConfig === 'string') {
     // Angular CLI can have a builder pointing to another package:builder
     const [packageName, executorName] = executorConfig.split(':');
-    return readExecutorJson(packageName, executorName, root, projects);
+    return readExecutorJson(packageName, executorName, root, projects, [
+      basePath,
+    ]);
   }
   const isNgCompat = !executorsJson.executors?.[executor];
   return { executorsFilePath, executorConfig, isNgCompat };

--- a/packages/nx/src/utils/plugins/plugin-capabilities.ts
+++ b/packages/nx/src/utils/plugins/plugin-capabilities.ts
@@ -182,6 +182,7 @@ export async function listPluginCapabilities(
       ...Object.keys(plugin.executors).map(
         (name) =>
           `${chalk.bold(name)} : ${resolveExecutorDescription(
+            pluginName,
             plugin.executors[name],
             projects
           )}`
@@ -204,20 +205,38 @@ export async function listPluginCapabilities(
 }
 
 function resolveExecutorDescription(
+  pluginName: string,
   executorJsonEntry: ExecutorsJsonEntry,
-  projects: Record<string, ProjectConfiguration>
+  projects: Record<string, ProjectConfiguration>,
+  requirePaths = getNxRequirePaths(workspaceRoot)
 ) {
   try {
     if (typeof executorJsonEntry === 'string') {
       // it points to another executor, resolve it
       const [pkgName, executor] = executorJsonEntry.split(':');
+      // read the package.json of the parent plugin
+      const { path: packageJsonPath } = readPluginPackageJson(
+        pluginName,
+        projects,
+        requirePaths
+      );
+      // accumulate the require paths to resolve nested packages
+      const cummulativeRequirePaths = [
+        ...requirePaths,
+        dirname(packageJsonPath),
+      ];
       const collection = loadExecutorsCollection(
-        workspaceRoot,
         pkgName,
-        projects
+        projects,
+        cummulativeRequirePaths
       );
 
-      return resolveExecutorDescription(collection[executor], projects);
+      return resolveExecutorDescription(
+        pkgName,
+        collection[executor],
+        projects,
+        cummulativeRequirePaths
+      );
     }
 
     return executorJsonEntry.description;
@@ -227,14 +246,14 @@ function resolveExecutorDescription(
 }
 
 function loadExecutorsCollection(
-  workspaceRoot: string,
   pluginName: string,
-  projects: Record<string, ProjectConfiguration>
+  projects: Record<string, ProjectConfiguration>,
+  requirePaths: string[]
 ): { [name: string]: ExecutorsJsonEntry } {
   const { json: packageJson, path: packageJsonPath } = readPluginPackageJson(
     pluginName,
     projects,
-    getNxRequirePaths(workspaceRoot)
+    requirePaths
   );
 
   return {

